src/communication/channel.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "channel.hpp"
+#include "log.hpp"
+#include "utils/aosheader.hpp"
+#include "utils/checksum.hpp"
+
+namespace aos::zephyr::communication {
+
+Mutex Channel::sWriteMutex;
+
+Channel::Channel(TransportItf* transport, int port)
```

Comment:
added another abstraction
```
/**
 * Communication interface.
 */
class CommunicationItf : public TransportItf { };
```

so instead of use in declaration`TransportItf` interface use `CommunicationItf`